backend/app/core/events_service.py

```
@@ -0,0 +1,187 @@
+"""Events service for handling complex event queries and business logic."""
+
+import logging
+from typing import List, Optional, Tuple
+
+from sqlalchemy import and_, func, or_, text
```

Ruff (0.12.2)

`sqlalchemy.and_` imported but unused

Remove unused import

(F401)